insecure/adversary/attacker_test.go

```diff
@@ -0,0 +1,214 @@
+package adversary_test
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	grpcinsecure "google.golang.org/grpc/credentials/insecure"
+
+	"github.com/onflow/flow-go/insecure"
+	"github.com/onflow/flow-go/insecure/adversary"
+	mockinsecure "github.com/onflow/flow-go/insecure/mock"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/model/libp2p/message"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/network"
+	"github.com/onflow/flow-go/network/codec/cbor"
+	"github.com/onflow/flow-go/utils/unittest"
+)
+
+const attackerAddress = "localhost:8000"
```

Comment:
Here is the example of the Access node test which allows dynamic ports instead of a fixed on - 
1. Test sets the port to 0 https://github.com/onflow/flow-go/blob/ceaa10f04aa3a05687dbc8197937c2f0a88f64d8/engine/access/rate_limit_test.go#L93-#L96
2. The engine code exposes a function retrieve the port https://github.com/onflow/flow-go/blob/3efe3a9375de4c61ff7568e3047ba640da835c3d/engine/access/rpc/engine.go#L244
3. The test waits for the port to be opened and then reads it - https://github.com/onflow/flow-go/blob/ceaa10f04aa3a05687dbc8197937c2f0a88f64d8/engine/access/rate_limit_test.go#L117-#L124

Reply:
Thanks, applied here: https://github.com/onflow/flow-go/pull/2121/commits/aed11d2e0c8cee21f550bd9cbe723229992e524a